### ISSUE TITLE
Bugsnag doesn't send the proper environment

### ISSUE THREAD

**Issue by deepj** [bugsnag/bugsnag-ruby#421]
### Expected behavior
Send the proper environment according what is set exactly in `RAILS_ENV` or `RACK_ENV`. 

### Observed behavior
Bugsnag sends `production` instead of `staging` even `staging` is set `RAILS_ENV` and `RACK_ENV` on Heroku

### Steps to reproduce
Our config:

```
Bugsnag.configure do |config|
  config.notify_release_stages = %w[production staging]
end
```

### Version
6.6.3

### Additional information
The application runs on Heroku

**Comment by Cawllec** [comment]
Hi @deepj, Bugsnag should automatically be setting the `release_stage` to the `Rails.env` or `RACK_ENV` environment variables during initialisation.  Have you checked that it isn't being overridden somewhere, and that your `BUGSNAG_RELEASE_STAGE` environment variable isn't set?

**Comment by adnan-panalytics** [comment]
@Cawllec I am also experiencing a similar problem. I have tried setting the RACK_ENV, RAILS_ENV and BUGSNAG_RELEASE_STAGE in the environment. It doesn't seem to pick any of them up. It marks all the bugs as Production (with a **P** in the overview)
Additionally, it seems to be happening only when an error occurs in a Resque job. When an error occurs on a server, it seems to be reported the right _STAGE_. Everything was replicated on my local machine (apart from the servers). 

If you need any further details, I can help with that.

**Comment by Cawllec** [comment]
Hi @adnan-panalytics, thanks for the info, I'll look into the resque/server issue, that sounds strange.  I assume you aren't running on Heroku?

**Comment by deepj** [comment]
@Cawllec Sorry for being late. Our problem was we use Grape without Rails as API backend. Se had to set RACK_ENV using `release_stage`. Without that, our `staging` environment on Heroku was ignored and it was identified as `production`.

**Comment by Cawllec** [comment]
Ok, thanks @deepj.

**Comment by adnan-panalytics** [comment]
@Cawllec No, we don't use Heroku. Our servers are deployed to AWS.

**Comment by phillipsam** [comment]
A fix for this has now been released in [v6.16.0](https://github.com/bugsnag/bugsnag-ruby/releases/tag/v6.16.0).
